x-pack/plugin/ml-package-loader/src/main/java/org/elasticsearch/xpack/ml/packageloader/action/DownloadTaskRemovedListener.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.ml.packageloader.action;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.support.master.AcknowledgedResponse;
+import org.elasticsearch.tasks.RemovedTaskListener;
+import org.elasticsearch.tasks.Task;
+
+public record DownloadTaskRemovedListener(ModelDownloadTask trackedTask, ActionListener<AcknowledgedResponse> listener)
+    implements
+        RemovedTaskListener {
+
+    @Override
+    public void onRemoved(Task task) {
+        if (task.getId() == trackedTask.getId() && task.getAction().equals(trackedTask.getAction())) {
```

Comment:
Isn't just the `id` enough to identify the task? Do you need to action to match?

Reply:
I wasn't sure. After checking the code I realised the task Id is generated by an `AtomicLong` and is enough for a safe match

https://github.com/elastic/elasticsearch/blob/a59c182f9f7e9d1bf3d6eecbc0e44f24ff91d053/server/src/main/java/org/elasticsearch/tasks/TaskManager.java#L144